File: internal/model/category.go
package model

type Category struct {
	CommonModel

	Name        string `gorm:"size:255;unique;not null;comment:分类名称" json:"name"`
	Slug        string `gorm:"size:255;uniqueIndex;not null;comment:分类 slug" json:"slug"`
	Description string `gorm:"size:255;comment:分类描述" json:"description"`

	Blogs []*Blog `gorm:"foreignKey:CategoryID" json:"blogs,omitempty"`
}

File: web/packages/fuxiaochen-types/src/category.ts
import { Blog } from "./blog";
import type { CommonModel, ListReq } from "./common";

export interface Category extends CommonModel {
	name: string;
	slug: string;
	description: string;
	blogCount: number;
	blogs?: Blog[];
}

export interface CategoryListReq extends ListReq {
	name?: string;
	slug?: string;
}

export interface CategoryListResp {
	total: number;
	lists: Category[];
}

export interface CategoryCreateReq {
	name: string;
	slug: string;
	description?: string;
}
